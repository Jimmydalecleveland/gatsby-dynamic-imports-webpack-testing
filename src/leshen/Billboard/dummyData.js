import lifestyle from '../../images/split-content-lifestyle.jpg'
import lifestyleFull from '../../images/billboard-lifestyle-full.jpg'

export const FLUID_IMAGE_DATA = {
  context: {
    custom: {
      alt: 'Alt Text',
    },
  },
  fluid: {
    base64:
      'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAPABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAQFAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAABAv/aAAwDAQACEAMQAAABY4VxKBIGf//EABoQAAIDAQEAAAAAAAAAAAAAAAABAgMREjL/2gAIAQEAAQUCrjpP1qE3lksOz//EABYRAQEBAAAAAAAAAAAAAAAAAAAREv/aAAgBAwEBPwHSv//EABURAQEAAAAAAAAAAAAAAAAAAAAR/9oACAECAQE/AYj/xAAcEAACAQUBAAAAAAAAAAAAAAAAATECEBJBUWH/2gAIAQEABj8C6JUySZ6F7b//xAAcEAACAgIDAAAAAAAAAAAAAAABEQAxIVEQQZH/2gAIAQEAAT8hIHQaigsQF4gXT2NWQVuEdhW3x//aAAwDAQACAAMAAAAQ4+//xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EMVUo//EABYRAQEBAAAAAAAAAAAAAAAAAAARUf/aAAgBAgEBPxDaH//EAB4QAQACAgEFAAAAAAAAAAAAAAERIQAxcUFRYZGx/9oACAEBAAE/EEKI60I86xyRZUEjqPvNJHwy9miG2LXgZRab+9sUtAHGf//Z',
    src: lifestyle,
    srcSet: lifestyle,
    aspectRatio: 1.3524904214559388,
    sizes: '(max-width: 706px) 100vw, 706px',
    presentationWidth: 2560,
  },
}

export const FLUID_IMAGE_DATA_FULL = {
  context: {
    custom: {
      alt: 'Alt Text',
    },
  },
  fluid: {
    base64:
      'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAGABQDASIAAhEBAxEB/8QAFgABAQEAAAAAAAAAAAAAAAAAAAMC/8QAFQEBAQAAAAAAAAAAAAAAAAAAAAH/2gAMAwEAAhADEAAAAdULJA//xAAZEAACAwEAAAAAAAAAAAAAAAABAgADBBL/2gAIAQEAAQUCTYTBa1kfXZ3/AP/EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8BP//EABURAQEAAAAAAAAAAAAAAAAAABAR/9oACAECAQE/AYf/xAAcEAACAQUBAAAAAAAAAAAAAAAAEQECEiExQXH/2gAIAQEABj8C0i6ZxTwlJeH/xAAZEAADAQEBAAAAAAAAAAAAAAABESEAQVH/2gAIAQEAAT8hN23nNcLWQCawQJjHn//aAAwDAQACAAMAAAAQBB//xAAWEQEBAQAAAAAAAAAAAAAAAAABEBH/2gAIAQMBAT8QXJ//xAAWEQEBAQAAAAAAAAAAAAAAAAABABH/2gAIAQIBAT8QA2X/xAAaEAEAAgMBAAAAAAAAAAAAAAABACERMWGR/9oACAEBAAE/EGQY0bPrcwQjrko72algtuO3P//Z',
    src: lifestyleFull,
    srcSet: lifestyleFull,
    aspectRatio: 3.116883116883117,
    sizes: '(max-width: 800px) 100vw, 800px',
    presentationWidth: 2560,
  },
}
