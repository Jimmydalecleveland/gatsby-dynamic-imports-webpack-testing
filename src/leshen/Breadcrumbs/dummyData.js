export const CRUMB_DATA = [
  {
    shortName: 'Home',
    path: '/',
  },
  {
    shortName: 'Internet',
    path: '/internet',
  },
  {
    shortName: 'Gigabit',
    path: '/gigabit',
  },
  {
    shortName: 'Gigabit Plans',
    path: '/gigabit-plans',
  },
]
