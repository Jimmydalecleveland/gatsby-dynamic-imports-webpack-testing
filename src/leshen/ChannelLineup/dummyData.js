const CHANNEL_LINEUP_DATA = {
  channels: {
    edges: [
      {
        node: {
          name: 'History Channel',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'history-channel',
        },
      },
      {
        node: {
          name: 'Nickelodeon ',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'nickelodeon',
        },
      },
      {
        node: {
          name: 'FX',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'fx',
        },
      },
      {
        node: {
          name: 'ESPN',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'espn',
        },
      },
      {
        node: {
          name: 'FOX',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'fox',
        },
      },
      {
        node: {
          name: 'CNN',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'cnn',
        },
      },
      {
        node: {
          name: 'Cartoon Network',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'cartoon-network',
        },
      },
      {
        node: {
          name: 'Comedy Central',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'comedy-central',
          url: '/tv/channels/comedy-central',
        },
      },
      {
        node: {
          name: 'Nickelodeon / Nick At Nite - West (NICKW)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'nickelodeon',
        },
      },
      {
        node: {
          name: 'Turner Network Television (TNT)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
          ],
          brandy_id: 'tnt',
          url: '/tv/channels/tnt',
        },
      },
      {
        node: {
          name: 'TV Land (TVLND)',
          packageAvailability: ['dn-americastop120+', 'dn-americastop250'],
          brandy_id: 'tvland',
        },
      },
      {
        node: {
          name: 'E! Entertainment Television (E!)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
          ],
          brandy_id: 'e',
        },
      },
      {
        node: {
          name: 'Syfy Channel (SYFY)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'syfy',
        },
      },
      {
        node: {
          name: 'Eternal Word Television Network (EWTN)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'ewtn',
        },
      },
    ],
  },
  packages: {
    edges: [
      {
        node: {
          name: "America's Top 120",
          brandy_id: 'dn-americastop120',
        },
      },
      {
        node: {
          name: "America's Top 120+",
          brandy_id: 'dn-americastop120+',
        },
      },
      {
        node: {
          name: "America's Top 200",
          brandy_id: 'dn-americastop200',
        },
      },
      {
        node: {
          name: "America's Top 250",
          brandy_id: 'dn-americastop250',
        },
      },
    ],
  },
}

export default CHANNEL_LINEUP_DATA
