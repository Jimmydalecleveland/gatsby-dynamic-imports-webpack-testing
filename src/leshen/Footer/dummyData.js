export const FOOTER_NAV_ITEMS = [
  {
    text: 'Services',
    href: false,
    subnav: [
      {
        text: 'Clearlink Internet',
        href: '#',
      },
      {
        text: 'FiOS Internet',
        href: '#',
      },
      {
        text: 'Bundles',
        href: '#',
      },
      {
        text: 'Frontier TV',
        href: '#',
      },
      {
        text: 'Dish TV',
        href: '#',
      },
      {
        text: 'Clearlink Phone',
        href: '#',
      },
      {
        text: 'Business',
        href: '#',
      },
    ],
  },
  {
    text: 'About',
    href: false,
    subnav: [
      {
        text: 'Store Locations',
        href: '#',
      },
      {
        text: 'Media Center',
        href: '#',
      },
      {
        text: 'Scholarship',
        href: '#',
      },
      {
        text: 'Contact Us',
        href: '#',
      },
    ],
  },
  {
    text: 'Español',
    href: false,
    subnav: [
      {
        text: 'Internet',
        href: '#',
      },
      {
        text: 'FiOS Internet',
        href: '#',
      },
      {
        text: 'DishLATINO',
        href: '#',
      },
      {
        text: 'Teléfono',
        href: '#',
      },
      {
        text: 'Paquetes',
        href: '#',
      },
    ],
  },
  {
    text: 'Terms',
    href: false,
    subnav: [
      {
        text: 'Legal',
        href: '#',
      },
      {
        text: 'Privacy Policy',
        href: '#',
      },
      {
        text: 'Terms of Service',
        href: '#',
      },
    ],
  },
]

export const DISCLAIMERS = [
  '©2019 Internet Service Partners. All Rights Reserved. The name CenturyLink and the pathways logo are trademarks of CenturyLink.',
  '*Geographic and service restrictions apply to AT&T services. Call to see if you qualify. Fiber optics may apply to all or part of the network, depending on your location.',
  '© 2019 AT&T Intellectual Property. AT&T, the AT&T logo and all other AT&T marks contained herein are trademarks of AT&T Intellectual Property and/or AT&T affiliated companies. All rights reserved. Subsidiaries and affiliates of AT&T Inc. provide products and services under the AT&T brand.',
]
