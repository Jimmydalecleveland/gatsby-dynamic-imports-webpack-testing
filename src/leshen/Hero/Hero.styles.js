import styled from "@emotion/styled"
import { css } from "@emotion/core"
import { stripUnit } from "polished"

import Image from "../Image"
import Section from "../Section"
import fluidRange from "../utils/fluidRange"

export const Hero = styled(Section)(
  ({ theme }) =>
    css`
      position: relative;
      z-index: 0;

      @media (min-width: ${theme.breakpoints.md}) {
        padding-right: 50%;
      }
    `
)

export const Content = styled.div(
  ({ split, theme }) => css`
    position: relative;
    z-index: 2;
    ${split &&
    fluidRange("padding-right", [
      stripUnit(theme.spacing.lg),
      stripUnit(theme.spacing.xxl),
    ])};
  `
)

export const BackgroundImage = styled(Image)(
  ({ theme, split }) => css`
    position: absolute !important;
    top: 0;
    left: ${split ? "50%" : 0};
    bottom: 0;
    right: 0;
    display: none;
    opacity: 0;

    @media (min-width: ${theme.breakpoints.md}) {
      display: block;
      opacity: 1;
    }
  `
)
