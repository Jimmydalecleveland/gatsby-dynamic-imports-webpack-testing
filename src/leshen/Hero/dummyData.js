import hero from "../images/hero.jpg"
import lifestyle from "../images/split-content-lifestyle.jpg"

export const FLUID_IMAGE_DATA = {
  context: {
    custom: {
      alt: "Alt Text",
    },
  },
  fluid: {
    base64:
      "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAJABQDASIAAhEBAxEB/8QAFwAAAwEAAAAAAAAAAAAAAAAAAAQFA//EABUBAQEAAAAAAAAAAAAAAAAAAAAB/9oADAMBAAIQAxAAAAG9IagJsJB//8QAHBAAAQMFAAAAAAAAAAAAAAAAAQACAwQRMTIz/9oACAEBAAEFAlVSNIMgu/o7Mu//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AT//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AT//xAAZEAEAAgMAAAAAAAAAAAAAAAABABASIYH/2gAIAQEABj8CgE3nxor/xAAaEAACAwEBAAAAAAAAAAAAAAAAAREhMXGh/9oACAEBAAE/IVg/i1cjso2x04eU0P/aAAwDAQACAAMAAAAQC8//xAAWEQEBAQAAAAAAAAAAAAAAAAAAARH/2gAIAQMBAT8QrH//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/ED//xAAcEAEAAQUBAQAAAAAAAAAAAAABABEhMWGhsfD/2gAIAQEAAT8QsS4pGC3QMLp1LO2KgBic7yfJtnEeT//Z",
    src: hero,
    srcSet: hero,
    aspectRatio: 2.3529411764705883,
    sizes: "(max-width: 800px) 100vw, 800px",
    presentationWidth: 800,
  },
}

export const FLUID_IMAGE_DATA_SPLIT = {
  context: {
    custom: {
      alt: "Alt Text",
    },
  },
  fluid: {
    base64:
      "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAPABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAQFAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAABAv/aAAwDAQACEAMQAAABY4VxKBIGf//EABoQAAIDAQEAAAAAAAAAAAAAAAABAgMREjL/2gAIAQEAAQUCrjpP1qE3lksOz//EABYRAQEBAAAAAAAAAAAAAAAAAAAREv/aAAgBAwEBPwHSv//EABURAQEAAAAAAAAAAAAAAAAAAAAR/9oACAECAQE/AYj/xAAcEAACAQUBAAAAAAAAAAAAAAAAATECEBJBUWH/2gAIAQEABj8C6JUySZ6F7b//xAAcEAACAgIDAAAAAAAAAAAAAAABEQAxIVEQQZH/2gAIAQEAAT8hIHQaigsQF4gXT2NWQVuEdhW3x//aAAwDAQACAAMAAAAQ4+//xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EMVUo//EABYRAQEBAAAAAAAAAAAAAAAAAAARUf/aAAgBAgEBPxDaH//EAB4QAQACAgEFAAAAAAAAAAAAAAERIQAxcUFRYZGx/9oACAEBAAE/EEKI60I86xyRZUEjqPvNJHwy9miG2LXgZRab+9sUtAHGf//Z",
    src: lifestyle,
    srcSet: lifestyle,
    aspectRatio: 1.3524904214559388,
    sizes: "(max-width: 706px) 100vw, 706px",
    presentationWidth: 706,
  },
}
