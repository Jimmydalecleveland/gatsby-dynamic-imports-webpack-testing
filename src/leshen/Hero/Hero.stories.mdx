import { Story, Meta, Preview, Props, Anchor } from '@storybook/addon-docs/blocks'
import { Done, Phone } from 'emotion-icons/material'

import Hero, { ticketID } from '.'
import Typography from '../Typography'
import PackageProvider from '../PackageProvider'
import List from '../List'
import ListItem from '../List/ListItem'
import Button from '../Button'
import Breadcrumbs from '../Breadcrumbs'
import { FLUID_IMAGE_DATA, FLUID_IMAGE_DATA_SPLIT } from './dummyData'
import { CRUMB_DATA } from '../Breadcrumbs/dummyData'
import ComponentProgress from '../ComponentProgress'

import HeroImage from '../../images/contentful-stories/hero.png'
import HeroFieldsImage from '../../images/contentful-stories/hero-fields.png'
import HeroPackageImage from '../../images/contentful-stories/hero-module-package.png'

<Meta title='Components|Hero' component={Hero} decorators={[(story) => <><ComponentProgress ticketID={ticketID} />{story()}</>]} />

# Hero

<Preview>
  <Story name="basic">
    <Hero image={FLUID_IMAGE_DATA}>
      <Typography variant="h1">
        Bundle DIRECTV Packages and CenturyLink
      </Typography>
      <Typography variant="h4" rendersAs="h2">
        DIRECTV SELECT All-included + CenturyLink Internet up to 20 Mbps
        starting at
      </Typography>
      <PackageProvider
        price={{ amount: '45', condensed: true, large: true }}
        bullets={
          <List>
            <ListItem icon={<Done />}>
              <Typography variant="h5" rendersAs="span">
                Download Speeds up to 20 Mbps
              </Typography>
            </ListItem>
            <ListItem icon={<Done />}>
              <Typography variant="h5" rendersAs="span">
                No Contract
              </Typography>
            </ListItem>
            <ListItem icon={<Done />}>
              <Typography variant="h5" rendersAs="span">
                No Bundling Required
              </Typography>
            </ListItem>
          </List>
        }
        legal={
          <p>
            Speed may not be available in your area. Rate excludes taxes;
            activation fee applies.
          </p>
        }
      />
      <Button leftIcon={<Phone />} variant="hero">
        Call Now 1.888.888.8888
      </Button>
    </Hero>
  </Story>
</Preview>

## Props
<Props of={Hero} />

## Split Variant
<Preview>
  <Story name="split variant">
    <Hero image={FLUID_IMAGE_DATA_SPLIT} type="Split" darkText={false}>
      <Typography variant="h1">
        Bundle DIRECTV Packages and CenturyLink
      </Typography>
      <Typography variant="h4" rendersAs="h2">
        DIRECTV SELECT All-included + CenturyLink Internet up to 20 Mbps
        starting at
      </Typography>
      <PackageProvider
        price={{ amount: '45', condensed: true, large: true }}
        bullets={
          <List>
            <ListItem icon={<Done />}>
              <Typography variant="h5" rendersAs="span">
                Download Speeds up to 20 Mbps
              </Typography>
            </ListItem>
            <ListItem icon={<Done />}>
              <Typography variant="h5" rendersAs="span">
                No Contract
              </Typography>
            </ListItem>
            <ListItem icon={<Done />}>
              <Typography variant="h5" rendersAs="span">
                No Bundling Required
              </Typography>
            </ListItem>
          </List>
        }
        legal={
          <p>
            Speed may not be available in your area. Rate excludes taxes;
            activation fee applies.
          </p>
        }
      />
      <Button leftIcon={<Phone />} variant="hero">
        Call Now 1.888.888.8888
      </Button>
    </Hero>
  </Story>
</Preview>

## With Breadcrumbs
<Preview>
  <Story name="with breadcrumbs">
    <Hero
      image={FLUID_IMAGE_DATA}
      breadcrumbProps={CRUMB_DATA}>
      <Typography variant="h1">
        Bundle DIRECTV Packages and CenturyLink
      </Typography>
      <Typography variant="h4" rendersAs="h2">
        DIRECTV SELECT All-included + CenturyLink Internet up to 20 Mbps
        starting at
      </Typography>
      <PackageProvider
        price={{ amount: '45', condensed: true, large: true }}
        bullets={
          <List>
            <ListItem icon={<Done />}>
              <Typography variant="h5" rendersAs="span">
                Download Speeds up to 20 Mbps
              </Typography>
            </ListItem>
            <ListItem icon={<Done />}>
              <Typography variant="h5" rendersAs="span">
                No Contract
              </Typography>
            </ListItem>
            <ListItem icon={<Done />}>
              <Typography variant="h5" rendersAs="span">
                No Bundling Required
              </Typography>
            </ListItem>
          </List>
        }
        legal={
          <p>
            Speed may not be available in your area. Rate excludes taxes;
            activation fee applies.
          </p>
        }
      />
      <Button leftIcon={<Phone />} variant="hero">
        Call Now 1.888.888.8888
      </Button>
    </Hero>
  </Story>
</Preview>

## Contentful

<Preview>
  <Story name="contentful">
    <>
      <Typography>In the "Page" entry, click "Create a new Hero and link". <strong>"Type", "Text Color", and "Heading" are required.</strong></Typography>
      <img src={HeroImage} alt="Example of a Hero" />
      <Typography>An example of a Hero entry's fields</Typography>
      <img src={HeroFieldsImage} alt="Example of a Hero entry's fields" />
      <Typography>A package may be added to the hero from the "Modules" entry.</Typography>
      <img src={HeroPackageImage} alt="Example of a Hero with a package" />
    </>
  </Story>
</Preview>
