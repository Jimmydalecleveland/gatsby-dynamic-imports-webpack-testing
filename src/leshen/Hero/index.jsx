import React from 'react'
import PropTypes from 'prop-types'

import * as Styled from './Hero.styles'
import Typography from '../Typography'
import Breadcrumbs from '../Breadcrumbs'

const Hero = ({
  heading,
  subheading,
  mainContent,
  image,
  children,
  headingLevel,
  breadcrumbProps,
  backgroundColor,
  ...props
}) => (
  <Styled.Hero backgroundColor={backgroundColor} {...props}>
    {breadcrumbProps && (
      <Breadcrumbs {...breadcrumbProps} backgroundColor={backgroundColor} />
    )}
    <Styled.Content>
      {heading && (
        <Typography variant={`h${headingLevel}`}>{heading}</Typography>
      )}
      {subheading && (
        <Typography
          variant={`h${headingLevel + 3}`}
          rendersAs={`h${headingLevel + 1}`}
        >
          {subheading}
        </Typography>
      )}
      {mainContent}
      {children}
    </Styled.Content>
    {image && <Styled.BackgroundImage data={image} maxWidth="none" />}
  </Styled.Hero>
)

Hero.defaultProps = {
  heading: '',
  subheading: '',
  mainContent: null,
  children: null,
  image: null,
  headingLevel: 2,
  backgroundColor: null,
  breadcrumbProps: null,
}

Hero.propTypes = {
  heading: PropTypes.string,
  subheading: PropTypes.string,
  mainContent: PropTypes.node,
  children: PropTypes.node,
  image: PropTypes.shape({
    fluid: PropTypes.object,
    fixed: PropTypes.object,
    svg: PropTypes.object,
    context: PropTypes.object,
  }),
  backgroundColor: PropTypes.string,
  breadcrumbProps: PropTypes.shape({
    crumbs: PropTypes.string,
    path: PropTypes.string,
    siteURL: PropTypes.string,
  }),
  headingLevel: PropTypes.oneOf([1, 2]),
}

export default Hero

export const ticketID = 'CMS-166'
