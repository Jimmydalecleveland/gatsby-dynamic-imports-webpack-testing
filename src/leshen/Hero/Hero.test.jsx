import React from 'react'

import { render } from 'setup-tests'
import { FLUID_IMAGE_DATA } from './dummyData'

import Hero from '.'

describe('Hero', () => {
  test('SplitContent should render', () => {
    const { container } = render(
      <Hero image={FLUID_IMAGE_DATA}>
        <p>Hero</p>
      </Hero>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
