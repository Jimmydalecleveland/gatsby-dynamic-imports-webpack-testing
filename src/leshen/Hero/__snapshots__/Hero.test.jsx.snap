// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero SplitContent should render 1`] = `
.emotion-6 {
  padding-left: calc(2.2222222222222223vw + 21.333333333333332px);
  padding-right: calc(2.2222222222222223vw + 21.333333333333332px);
  padding-top: calc(7.777777777777778vw + -5.333333333333333px);
  padding-bottom: calc(7.777777777777778vw + -5.333333333333333px);
  position: relative;
  position: relative;
  z-index: 0;
}

@media (max-width:480px) {
  .emotion-6 {
    padding-left: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-6 {
    padding-left: 48px;
  }
}

@media (max-width:480px) {
  .emotion-6 {
    padding-right: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-6 {
    padding-right: 48px;
  }
}

@media (min-width:1288px) {
  .emotion-6 {
    padding-right: calc((100% - 1200px) / 2);
    padding-left: calc((100% - 1200px) / 2);
  }
}

@media (max-width:480px) {
  .emotion-6 {
    padding-top: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-6 {
    padding-top: 88px;
  }
}

@media (max-width:480px) {
  .emotion-6 {
    padding-bottom: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-6 {
    padding-bottom: 88px;
  }
}

@media (min-width:768px) {
  .emotion-6 {
    padding-right: 50%;
  }
}

.emotion-0 {
  position: relative;
  z-index: 2;
}

.emotion-3 {
  max-width: none;
  width: 100%;
  margin: 0 auto;
  position: absolute !important;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  display: none;
  opacity: 0;
}

@media (min-width:768px) {
  .emotion-3 {
    display: block;
    opacity: 1;
  }
}

<section
  class="emotion-5 emotion-6 emotion-7"
>
  <div
    class="emotion-0 emotion-1"
  >
    
    
    <p>
      Hero
    </p>
  </div>
  <div
    class="emotion-2 emotion-3 emotion-4 gatsby-image-wrapper"
    style="position: relative; overflow: hidden;"
  >
    <div
      aria-hidden="true"
      style="width: 100%; padding-bottom: 42.5%;"
    />
    <img
      alt=""
      aria-hidden="true"
      src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAJABQDASIAAhEBAxEB/8QAFwAAAwEAAAAAAAAAAAAAAAAAAAQFA//EABUBAQEAAAAAAAAAAAAAAAAAAAAB/9oADAMBAAIQAxAAAAG9IagJsJB//8QAHBAAAQMFAAAAAAAAAAAAAAAAAQACAwQRMTIz/9oACAEBAAEFAlVSNIMgu/o7Mu//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AT//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AT//xAAZEAEAAgMAAAAAAAAAAAAAAAABABASIYH/2gAIAQEABj8CgE3nxor/xAAaEAACAwEBAAAAAAAAAAAAAAAAAREhMXGh/9oACAEBAAE/IVg/i1cjso2x04eU0P/aAAwDAQACAAMAAAAQC8//xAAWEQEBAQAAAAAAAAAAAAAAAAAAARH/2gAIAQMBAT8QrH//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/ED//xAAcEAEAAQUBAQAAAAAAAAAAAAABABEhMWGhsfD/2gAIAQEAAT8QsS4pGC3QMLp1LO2KgBic7yfJtnEeT//Z"
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
    />
    <picture>
      <source
        sizes="(max-width: 800px) 100vw, 800px"
        srcset="test-file-stub"
      />
      <img
        alt="Alt Text"
        loading="lazy"
        sizes="(max-width: 800px) 100vw, 800px"
        src="test-file-stub"
        srcset="test-file-stub"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
      />
    </picture>
    <noscript>
      &lt;picture&gt;&lt;source srcset="test-file-stub" sizes="(max-width: 800px) 100vw, 800px" /&gt;&lt;img loading="lazy" sizes="(max-width: 800px) 100vw, 800px" srcset="test-file-stub" src="test-file-stub" alt="Alt Text" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
  </div>
</section>
`;
